### ISSUE TITLE
render function is not pure

### ISSUE THREAD

**Issue by aaronshaf** [aaronshaf/react-toggle#18]
The render function calls `isChecked`, which has:

``` javascript
if (this.refs.input) {
    return this.refs.input.getDOMNode().checked
}
```

React then warns:

> "Warning: PublishToggle is accessing getDOMNode or findDOMNode inside its render(). render() should be a pure function of props and state. It should never access something that requires stale data from the previous render, such as refs. Move this logic to componentDidMount and componentDidUpdate instead."

**Comment by aaronshaf** [comment]
^ @brentburg

**Comment by jedwards1211** [comment]
Here's how I'm thinking we can solve this issue:
- Implement `componentWillReceiveProps` to set `this.state.checked` from the props
- Get rid of `isChecked()`; use `this.state.checked` instead
- If `this.props.checked` is defined (even if falsey):
  - `handleClick` should call `this.props.onChange` if it is defined.  `handleClick` shouldn't do anything else, especially not trigger a re-render; if `this.props.onChange` sets `this.props.checked` to a different value, it will trigger a re-render automatically
- If `this.props.checked` is undefined:
  - `handleClick` should simulate a click on the checkbox, and call `setState()` to set `this.state.checked` to the new value of `checkbox.checked`

**Comment by ahstro** [comment]
Closing this, as it seems to have been fixed in fb91f14dc6a31729275eeb8bb3ced42fd4f20a3f
